Added Web App Example For Using Data Protection Service (#28803)
dotnet/AspNetCore.Docs: aspnetcore/security/data-protection/consumer-apis/overview.md

3. Creating an `IDataProtector` from an `IDataProtectionProvider` and using it to protect and unprotect data.

> ## Console Application Example

Reviewer: ```suggestion
## Console app
```

[!code-csharp[](../using-data-protection/samples/protectunprotect.cs?highlight=26,34,35,36,37,38,39,40)]

## Web app

> 

Reviewer: ```suggestion
Call <xref:Microsoft.Extensions.DependencyInjection.DataProtectionServiceCollectionExtensions.AddDataProtection(Microsoft.Extensions.DependencyInjection.IServiceCollection,System.Action{Microsoft.AspNetCore.DataProtection.DataProtectionOptions})> in `Program.cs`:

```

> ## Web Application Example

Reviewer: ```suggestion
## Web app
```

[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample1.cs?highlight=5)]

> 

Reviewer: ```suggestion

The following highlighted code shows how to use <xref:Microsoft.AspNetCore.DataProtection.IDataProtector> in a controller:

```

[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample2.cs?highlight=3,4,5,6,7,8,18,22)]

> The package `Microsoft.AspNetCore.DataProtection.Abstractions` contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage:

Reviewer: ```suggestion
The package `Microsoft.AspNetCore.DataProtection.Abstractions` contains an extension method <xref:Microsoft.AspNetCore.DataProtection.DataProtectionCommonExtensions.GetDataProtector%2A> as a developer convenience. It encapsulates as a single operation both retrieving an <xref:Microsoft.AspNetCore.DataProtection.IDataProtectionProvider> from the service provider and calling [`IDataProtectionProvider.CreateProtector`](xref:Microsoft.AspNetCore.DataProtection.IDataProtectionProvider.CreateProtector%2A). The following sample demonstrates its usage:
```

3. Creating an `IDataProtector` from an `IDataProtectionProvider` and using it to protect and unprotect data.

## Console Application Example

[!code-csharp[](../using-data-protection/samples/protectunprotect.cs?highlight=26,34,35,36,37,38,39,40)]

## Web Application Example

> [!code-csharp[](../using-data-protection/samples/webappexample1.cs?highlight=5)]
> 
> [!code-csharp[](../using-data-protection/samples/webappexample2.cs?highlight=3,4,5,6,7,8,18,22)]

Reviewer: ```suggestion
[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample1.cs?highlight=5)]

[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample2.cs?highlight=3,4,5,6,7,8,18,22)]
```

> > [!NOTE]
> > The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.

Reviewer: ```suggestion
The package `Microsoft.AspNetCore.DataProtection.Abstractions` contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage:
```
